backend/base.py

```diff
@@ -5,51 +5,52 @@
                                unset_jwt_cookies, jwt_required, JWTManager
 
 
-api = Flask(__name__)
-
-api.config["JWT_SECRET_KEY"] = "SuperSecret"
-api.config["JWT_ACCESS_TOKEN_EXPIRES"] = timedelta(hours=1)
-jwt = JWTManager(api)
-
-@api.after_request
-def refresh_expiring_jwts(response):
-    try:
-        exp_timestamp = get_jwt()["exp"]
-        now = datetime.now(timezone.utc)
-        target_timestamp = datetime.timestamp(now + timedelta(minutes=30))
-        if target_timestamp > exp_timestamp:
-            access_token = create_access_token(identity=get_jwt_identity())
-            data = response.get_json()
-            if type(data) is dict:
-                data["access_token"] = access_token 
-                response.data = json.dumps(data)
-        return response
-    except (RuntimeError, KeyError):
-        return response
-
-@api.route('/token', methods=["POST"])
-def create_token():
-    email = request.json.get("email", None)
-    password = request.json.get("password", None)
-    if email != "test" or password != "test":
-        return {"msg": "Wrong email or password"}, 401
-
-    access_token = create_access_token(identity=email)
-    response = {"access_token":access_token}
-    return response
-
-@api.route("/logout", methods=["POST"])
-def logout():
-    response = jsonify({"msg": "logout successful"})
-    unset_jwt_cookies(response)
-    return response
-
-@api.route('/profile')
-@jwt_required()
-def my_profile():
+api = Flask(__name__) # Initilizing a flask web app names api
+
+api.config["JWT_SECRET_KEY"] = "SuperSecret" # SeTs key to encode and decode tokens
+api.config["JWT_ACCESS_TOKEN_EXPIRES"] = timedelta(hours=1) # Sets the expiration time for tokens for 1 hour
+jwt = JWTManager(api) # creates instance of JWTManager directly for this flak application and assigns it to the variable jwt
+
+@api.after_request # after any request run code below
+def refresh_expiring_jwts(response): # defines function to be run after a request
+    try: # attempts code below
+        exp_timestamp = get_jwt()["exp"] # sets exp_timestamp to the expiration timestamp of the access token
+        now = datetime.now(timezone.utc) # sets variable now to the current time
+        target_timestamp = datetime.timestamp(now + timedelta(minutes=30)) # sets target_timestamp to 30 minutes from current time
+        if target_timestamp > exp_timestamp: #if the target is futher in the future than the current expiration, run the code below
+            access_token = create_access_token(identity=get_jwt_identity()) # makes a new access token with the same user information
+            data = response.get_json() # sets data to the json response
+            if type(data) is dict: # checks to make sure the json data was retreived correctly by checking data type
```

Comment:
Json isn't only a dict. It could be a simple string or a list. But if it's either of those it means that it shouldn't try to attach and access_token.